regression-tests/sparktkregtests/testcases/models/arx_test.py

```diff
@@ -0,0 +1,110 @@
+        """Test ARx train method"""
+        self.assertEqual(500000, self.frame.count(), msg= "Dataframe was not created correctly. Check dataset does not include infinite")
+        train_frame = self.frame.copy(where= lambda row: row.Int <= 499990)
+        ts_column ="output"
+        x_columns = ["exo1", "exo2", "exo3"]
+
+        model = self.context.models.timeseries.arx.train(train_frame, ts_column, x_columns, 2, 0, False)
+
+        #These values are copied from the arx datagen file.
+        self.assertAlmostEqual(model.c, 1.335, delta=0.00000001)
+        self.assertAlmostEqual(model.coefficients[0], 0.542, delta=0.00000001)
+        self.assertAlmostEqual(model.coefficients[1], 0.237, delta=0.00000001)
+        self.assertAlmostEqual(model.coefficients[2], 0.1293, delta=0.00000001)
+        self.assertAlmostEqual(model.coefficients[3], 0.0781, delta=0.00000001)
+        self.assertAlmostEqual(model.coefficients[4], -0.04275, delta=0.00000001)
+
+    def test_arx_train_withErr(self):
+        """Test ARx train method with error introduced in dataset"""
+        self.assertEqual(500000, self.frame.count(), msg= "Dataframe was not created correctly. Check dataset does not include infinite")
+        train_frame = self.frame.copy(where= lambda row: row.Int <= 499990)
+        ts_column ="outputWithErr"
+        x_columns = ["exo1", "exo2", "exo3"]
+
+        model = self.context.models.timeseries.arx.train(train_frame, ts_column, x_columns, 2, 0, False)
+
+        #These values are copied from the arx datagen file.
+        self.assertAlmostEqual(model.c, 1.335, delta=0.001)
+        self.assertAlmostEqual(model.coefficients[0], 0.542, delta=0.001)
+        self.assertAlmostEqual(model.coefficients[1], 0.237, delta=0.001)
+        self.assertAlmostEqual(model.coefficients[2], 0.1293, delta=0.001)
+        self.assertAlmostEqual(model.coefficients[3], 0.0781, delta=0.001)
+        self.assertAlmostEqual(model.coefficients[4], -0.04275, delta=0.001)
+
+    def test_arx_predict(self):
+        """Test ARx predict method"""
+        self.assertEqual(500000, self.frame.count(), msg= "Dataframe was not created correctly. Check dataset does not include infinite")
+        train_frame = self.frame.copy(where= lambda row: row.Int <= 499990)
+        actual_data = self.frame.copy(where= lambda row: row.Int > 499990) #last 10 rows
+        ts_column ="output"
+        x_columns = ["exo1", "exo2", "exo3"]
+
+        model = self.context.models.timeseries.arx.train(train_frame, ts_column, x_columns, 2, 0, False)
+        predict_frame = model.predict(actual_data, ts_column, x_columns)
+        predict_data = predict_frame.take(n=actual_data.count(), columns="predicted_y")
+        prediction = [item for sublist in predict_data for item in sublist]
+
+        expected_data = actual_data.take(n=actual_data.count(), columns=ts_column)
+        expected_prediction = [item for sublist in expected_data for item in sublist]
+
+        for i in range(2,9):
```

Comment:
the first two terms are always useless, we don't expect them to match 
9 is hardcoded because everything about length of the frame is hardcoded